### DISCUSSION TITLE
Thank you for this library

### DISCUSSION THREAD

**Discussion by DirtyHairy** [HomeSpan/HomeSpan#232] [Show and tell]
Just wanted to give you a 👍 for this library. From what I gather implementing HAP is a nasty piece of work, and you did a great job. I am [using](https://github.com/DirtyHairy/rfkit) it to control 433MHz RF sockets from an ESP (together with a web interface for management and configuration), and HomeSpan has been very easy to set up and has been doing its job flawlessly from the start.

**Comment by HomeSpan** [comment]
Thanks very much for the kind feedback and I'm really glad you've been able to make good use of the library!  Good to know the library can be integrated as part of a larger ESP-IDF project (instead of used through the just the Arduino-IDE).

**Comment by DirtyHairy** [comment]
Yeah, I started out with PlatformIO, but I encountered a roadblock when the device would stop accepting new connections after a day or so. I spent some time tuning that maximum number of open sockets, but the issue still recurred, so I figured I might just as well be seeing a bug in LWIP or the ESP-IDF HTTP server (PlatformIO is still on ESP_IDF 3 / Arduinio 1) and switched to IDF.

I  haven't looked back 😏 The bug is gone, and I can finally tune the SDK config to my liking.

**Comment by HomeSpan** [comment]
That's great.  I noticed that in the last Arduino-ESP32 2.0.2 update (which uses the latest version 4 of the IDF), the maximum number of LWIP connections was (finally!) increased from 10 to 16.  This makes a big difference and I increased the corresponding number in an upcoming patch version of HomeSpan (max connections from 8 to 14, since 2 are needed in reserve).  If you are running a web server and/or OTA, you need reduce the max connections accordingly since they need LWIP socks as well.

**Comment by DirtyHairy** [comment]
> If you are running a web server and/or OTA, you need reduce the max connections accordingly since they need LWIP socks as well.

I had figured that much, and I had reduced the number of HomeKit sockets to 5 and the number of sockets of the server to 2 (I think it also uses an internal UDP socket), but the issue still persisted. I am now using 16 LWIP connections max, with 8 assigned to HomeKit, and 4 assigned to the server (I was a bit reluctant to go up to 14 sockets + 2 reserve) and it works fine. I might retry with 10 sockets assigned to HomeKit, but even with 8 it  handles juggling our various iThings very well.

**Comment by thncode** [comment]
@HomeSpan is there a coffee piggy bank address?

**Comment by HomeSpan** [comment]
@thncode, really appreciate the offer but the fact that lots of people are using the system, posting questions, providing great suggestions for new features/improvements, helping other users, etc., is the best contribution of all  😄.

**Reply by thncode** [reply]
KUDOS!
